patches/components-permissions-android-java-src-org-chromium-components-permissions-PermissionDialogDelegate.java.patch

```diff
@@ -0,0 +1,13 @@
+diff --git a/components/permissions/android/java/src/org/chromium/components/permissions/PermissionDialogDelegate.java b/components/permissions/android/java/src/org/chromium/components/permissions/PermissionDialogDelegate.java
+index d106ce2fcd624aaa2f52c6c1286ead73218eea0b..6ccbc043000bbdc1bafe61518227a104b7c4946e 100644
+--- a/components/permissions/android/java/src/org/chromium/components/permissions/PermissionDialogDelegate.java
++++ b/components/permissions/android/java/src/org/chromium/components/permissions/PermissionDialogDelegate.java
+@@ -18,7 +18,7 @@ import org.chromium.ui.base.WindowAndroid;
+  * respond appropriately.
+  */
+ @JNINamespace("permissions")
+-public class PermissionDialogDelegate {
++public class PermissionDialogDelegate extends BravePermissionDialogDelegate {
```

Comment:
this can be done with ASM cc @samartnik

Reply:
here we extend JNI calls, so it won't be compilable prior to ASM manipulations

Reply:
I just tried, JNI is not a problem, but there is still issue with casting object in `addLifetimeOptions`. We still can do it via ASM, but it's not that straightforward. I'll create a follow up issue for this.

Reply:
https://github.com/brave/brave-browser/issues/15243

Reply:
made all required java modifications via bytecode patches.